### PR TITLE
test: add comments and unittest for integral.h

### PR DESCRIPTION
**Pull request by pxlxingliang**
range: source/module_base

### CODE REVIEW

File: source/module_base/math_integral.h [hunk 1]
```diff
@@ -13,7 +13,14 @@ class Integral
     Integral();
     ~Integral();
 
-	// Peize Lin accelerate 2017-10-02
+    /**
+     * @brief simpson integral. Peize Lin accelerate 2017-10-02
```
Comment (caic99) [inline source/module_base/math_integral.h]: Author and date should be placed in a dedicated field.

File: source/module_base/test/math_integral_test.cpp [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+    void TearDown()
+    {
+        delete func;
+        delete rab;
+        delete asumlist;
+    }
+};
+
+TEST_F(Simpson_Integral_sinx,Simpson_Integral_rab)
```
Comment (caic99) [inline source/module_base/test/math_integral_test.cpp]: Both arguments TestFixtureName and TestName must be valid C++ identifiers and must not contain underscores (_). See https://google.github.io/googletest/reference/testing.html#TEST_F . 
@pxlxingliang Your former PRs also have the same problem. Sorry for my ignorance, it's my fault. Would you help me fixing them in another PR and renew test document? I suggest upper camel case is better (e.g. "SimpsonIntegralSinx").

File: source/module_base/test/math_integral_test.cpp [hunk 3]
```diff
@@ -0,0 +1,93 @@
+#include"../math_integral.h"
+#include"gtest/gtest.h"
+
+#include<math.h>
+
+#define PI 3.14159265358979
```
Comment (caic99) [inline source/module_base/test/math_integral_test.cpp]: Macro `M_PI` is included in `math.h`.

**Review by caic99 — CHANGES_REQUESTED** [review]

File: source/module_base/math_integral.h (hunk 1 above)
Comment (pxlxingliang) [reply source/module_base/math_integral.h]: Where is the dedicated field. How about behind the function name?
image

File: source/module_base/test/math_integral_test.cpp (hunk 2 above)
Comment (pxlxingliang) [reply source/module_base/test/math_integral_test.cpp]: ok, I will modify it. But I don't know the reason of not containing underscores, and It seems has no any impact on the previous using of unit test.

Comment (pxlxingliang) [reply source/module_base/test/math_integral_test.cpp]: gtest already has the explanation for this.
http://google.github.io/googletest/faq.html#why-should-test-suite-names-and-test-names-not-contain-underscore

Comment (caic99) [reply source/module_base/test/math_integral_test.cpp]: > ok, I will modify it. But I don't know the reason of not containing underscores, and It seems has no any impact on the previous using of unit test.

It's just conforms to C++ standard, and theoretically there would be no impact.

File: source/module_base/math_integral.h (hunk 1 above)
Comment (caic99) [reply source/module_base/math_integral.h]: > Where is the dedicated field. How about behind the function name? image

"author" and "date". Read more at [doxygen manual](https://www.doxygen.nl/manual/commands.html).

That's OK!

Comment (caic99) [reply source/module_base/math_integral.h]: @pxlxingliang And, I suggest you paste code in reply, but not screenshot.

**Review by caic99 — APPROVED** [review]

**Comment by pxlxingliang** [comment]
deleted by mistake. reopen it

**Review by caic99 — CHANGES_REQUESTED** [review]
Please follow up changes in #676 .

**Review by caic99 — APPROVED** [review]

**Pull request merged**